agents/ppo.py

```diff
@@ -4,110 +4,116 @@
-            ratio_clip=config.get('ratio_clip', 0.2),
             actor_loss_weight=config.get('actor_loss_weight', 1.0),
             critic_loss_weight=config.get('critic_loss_weight', 1.0),
-            bag_trajectory_memory=BaseBagTrajectoryMemory.create_from_config(config['path_memory']),
-            ppo_input_adapter=BaseInputAdapter.create_from_config(config['input_adapter']),
-            trajectory_sample_size=config.get('trajectory_sample_size', 100),
-            trajectory_length=config.get('trajectory_length', 100)
+            discount_factor=config.get('discount_factor', 0.99),
+            ratio_clip=config.get('ratio_clip', 0.99),
+            bag_trajectory_memory=BaseBagTrajectoryMemory.create_from_config(config['path_memory'])
+            if 'path_memory' in config else None,
+            node_idx=config.get('node_idx', None),
+            trajectory_length=config.get('trajectory_length', 100),
+            trajectory_sample_size=config.get('trajectory_sample_size', 100)
         )
 
-    def forward(self, inputs: Dict[str, Any]) -> Tensor:
-        bag_id, node_idx, neighbour, destination, storage = self._ppo_input_adapter.get_input(inputs)
-        next_node_idx, policy = self._actor(
-            node_idx=node_idx,
-            neighbour=neighbour,
-            destination=destination,
-            storage=storage
-        )
-        v_func = self._critic(
-            node_idx=node_idx,
-            destination=destination,
-            storage=storage
+    def forward(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+        # Shape: [batch_size]
+        current_node_idx = inputs[self._current_node_idx_prefix]
+        # Shape: [batch_size]
+        destination_node_idx = inputs[self._destination_node_idx_prefix]
+        # TensorWithMask
+        neighbour_node_ids = inputs[self._neighbours_node_ids_prefix]
+
+        # Shape: [batch_size], [batch_size, max_neighbors_num]
+        next_neighbors_ids, neighbors_logits = self._actor(
+            current_node_idx=current_node_idx,
+            neighbor_node_ids=neighbour_node_ids,
+            destination_node_idx=destination_node_idx
         )
-        self._bag_trajectory_memory.add_to_trajectory(
-            bag_ids=bag_id,
-            node_idxs=node_idx,
-            extra_infos=zip(v_func, policy, bag_id, node_idx, neighbour, destination, storage)
+
+        # Shape: [batch_size]
+        current_state_value_function = self._critic(
+            current_node_idx=current_node_idx,
+            destination_node_idx=destination_node_idx
         )
-        return next_node_idx
+
+        if self._bag_trajectory_memory is not None:
+            # TODO[Vladimir Baikalov]: Think about how to generalize
+            # Shape: [batch_size] if exists, None otherwise
+            bag_ids = inputs.get(self._bag_idx_prefix, None)
+            # Static entity if exists, None otherwise
+            storage = inputs.get(self._storage_prefix, None)
```

Comment:
It is still needed because we use it below to save trajectory in the memory. It is correct for `bag_ids`, I will remove `storage` since it's unused, thanks

Reply:
Yes, you are right. I will remove `storage` but still we need `bag_ids` for storing the trajectory